django/db/backends/oracle/base.py

```diff
@@ -475,23 +468,17 @@ def _fix_for_params(self, query, params, unify_by_values=False):
             params_dict = {param: ':arg%d' % i for i, param in enumerate(set(params))}
             args = [params_dict[param] for param in params]
             params = dict(zip(params_dict.values(), params_dict.keys()))
-            query = convert_unicode(query % tuple(args), self.charset)
+            query = force_text(query % tuple(args), self.charset)
         else:
             # Handle params as sequence
             args = [(':arg%d' % i) for i in range(len(params))]
-            query = convert_unicode(query % tuple(args), self.charset)
+            query = force_text(query % tuple(args), self.charset)
         return query, self._format_params(params)
```

Comment:
The issue happens when there's both a `BinaryField` and `TextField` in the same model. I guess the empty string values for insertion to those columns are being treated as equal when they need to be treated as different.